src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -401,6 +355,52 @@ bool ProcessPrimaryQueue ()
 			return true;
 		}
 
+		bool ProcessMarkedPending ()
+		{
+			bool marked = false;
+			foreach (var pending in Annotations.MarkedPending ()) {
+				var assembly = GetAssemblyFromMetadataTokenProvider (pending);
+				var assemblyAction = Annotations.GetAction (assembly);
+				if (assemblyAction == AssemblyAction.Skip)
+					continue;
+
+				marked = true;
+
+				// Some pending items might be processed by the time we get to them.
+				if (Annotations.IsProcessed (pending))
+					continue;
+
+				switch (pending) {
+				case TypeDefinition type:
+					MarkType (type, DependencyInfo.AlreadyMarked, type);
+					Debug.Assert (!Annotations.IsMarkedPending (type));
```

Comment:
I think these asserts are not adding value

Reply:
Removed